Code_Exercises/Exercise_04_Handling_Errors/source.cpp

```diff
@@ -13,29 +13,19 @@
 
 TEST_CASE("handling_errors", "handling_errors_source") {
 
-  // This program computes r = a + b
-  int a = 18, b = 24, r = 0;
-
-
   // Task: catch synchronous and asynchronous exceptions
+  
+  auto defaultQueue = sycl::queue{};
 
-  auto defaultQueue = sycl::queue{asyncHandler};
-
-  {
-    auto bufA = sycl::buffer{&a, sycl::range{1}};
-    auto bufB = sycl::buffer{&b, sycl::range{1}};
-    auto bufR = sycl::buffer{&r, sycl::range{1}};
+  auto buf = sycl::buffer<int>(sycl::range{1});
 
-    defaultQueue
-        .submit([&](sycl::handler& cgh) {
-          auto accA = sycl::accessor{bufA, cgh, sycl::read_only};
-          auto accB = sycl::accessor{bufB, cgh, sycl::read_only};
-          auto accR = sycl::accessor{bufR, cgh, sycl::write_only};
+  defaultQueue.submit([&](sycl::handler& cgh) {
+    // This throws an exception: an accessor has a range which is
```

Comment:
This is a nit but usually in the `source.cpp` we don't provide a solution. It might be more appropriate to say something like:

```
// Try to do something that might cause an asynchronous error to be thrown
// ie out of bound access, invalid ranged accessor, assert(false) within a kernel etc
```

And maybe set up the kernel as well with `cgh.single_task` or something